File: consumer/ios/UTS/UtsModule.m
//
//  UTSModule.m
//  Nammayatri
//
//  Created by Praveen kumar on 05/04/25.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(UTSModule, NSObject)

RCT_EXTERN_METHOD(requestBooking:(NSDictionary *)params
                 accessToken:(NSString *)accessToken
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(showTicket:(NSDictionary *)params
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(requestReinitialization:(NSDictionary *)params
                 accessToken:(NSString *)accessToken
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: consumer/ios/UTS/UtsModule.swift
//
//  UTSModule.swift
//  Nammayatri
//
//  Created by Praveen kumar on 05/04/25.
//

import Foundation
import UTS_SDK

@objc(UTSModule)
class UTSModule: NSObject {

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }

  let utsSdk: UTSSdk = UTSSdk();

  @objc(requestBooking:accessToken:withResolver:withRejecter:)
  func requestBooking(params: [String: Any], accessToken: String,
                resolve: @escaping RCTPromiseResolveBlock,
                reject: @escaping RCTPromiseRejectBlock) -> Void {
    utsSdk.requestBooking(
      params: params,
      accessToken: accessToken,
      onSuccess: { successMessage in
        let response: [String: Any] = [
          "status": "success",
          "data": successMessage
        ]
        resolve(response)
      },
      onError: { errorMessage in
        let response: [String: Any] = [
          "status": "error",
          "data": errorMessage
        ]
        resolve(response)
      }
    )
  }

  @objc(showTicket:withResolver:withRejecter:)
  func showTicket(params: [String: Any],
                     resolve: @escaping RCTPromiseResolveBlock,
                     reject: @escaping RCTPromiseRejectBlock) -> Void {
    utsSdk.showTicket(
      params: params,
      onSuccess: { successMessage in
        let response: [String: Any] = [
          "status": "success",
          "data": successMessage
        ]
        resolve(response)
      },
      onError: { errorMessage in
        let response: [String: Any] = [
          "status": "error",
          "data": errorMessage
        ]
        resolve(response)
      }
    )
  }

  @objc(requestReinitialization:accessToken:withResolver:withRejecter:)
  func requestReinitialization(params: [String: Any], accessToken: String,
                resolve: @escaping RCTPromiseResolveBlock,
                reject: @escaping RCTPromiseRejectBlock) -> Void {
    utsSdk.requestReinitialization(
      params: params,
      accessToken: accessToken,
      onSuccess: { successMessage in
        let response: [String: Any] = [
          "status": "success",
          "data": successMessage
        ]
        resolve(response)
      },
      onError: { errorMessage in
        let response: [String: Any] = [
          "status": "error",
          "data": errorMessage
        ]
        resolve(response)
      }
    )
  }
}
